cmd/surf/commands/parallelfilesprocessor.go

```diff
@@ -0,0 +1,84 @@
+package commands
+
+import (
+	"context"
+	"errors"
+	"github.com/CloudHub360/ch360.go/pool"
+	"github.com/mattn/go-zglob"
+)
+
+var ErrGlobMatchesNoFiles = errors.New("file pattern does not match any files")
+
+type ParallelFilesProcessor struct {
+	ProgressHandler ProgressHandler
+}
+
+//go:generate mockery -name "ProcessorFuncFactory"
+type ProcessorFuncFactory interface {
+	ProcessorFor(ctx context.Context, filename string) pool.ProcessorFunc
+}
+
+func (p *ParallelFilesProcessor) RunWithGlob(ctx context.Context,
+	filesPattern string,
+	parallelism int,
+	processorFuncFactory ProcessorFuncFactory) error {
+
+	files, _ := zglob.Glob(filesPattern)
+	fileCount := len(files)
+	if fileCount == 0 {
+		return ErrGlobMatchesNoFiles
+	}
+
+	return p.Run(ctx, files, parallelism, processorFuncFactory)
+}
+
+func (p *ParallelFilesProcessor) Run(ctx context.Context,
```

Comment:
❓ Can this be private? I was confused about what the public interface to this struct was, but I _think_ it is just `RunWithGlob`?

Reply:
I exposed this method for the tests, and because it seemed like a sensible signature regardless!